apps/meteor/app/file-upload/server/lib/FileUpload.ts

```diff
@@ -308,9 +308,18 @@ export const FileUpload = {
 		const store = FileUpload.getStore('Uploads');
 		const image = await store._store.getReadStream(file._id, file);
 
-		const transformer = sharp().resize({ width, height, fit: 'inside' });
-
-		const result = transformer.toBuffer({ resolveWithObject: true }).then(({ data, info: { width, height } }) => ({ data, width, height }));
+		let transformer = sharp().resize({ width, height, fit: 'inside' });
+
+		if (file.type === 'image/svg+xml') {
+			transformer = transformer.png();
+			file.name = (file.name || '').replace(/\.svg$/i, '.png');
+		}
+		const result = transformer.toBuffer({ resolveWithObject: true }).then(({ data, info: { width, height, format } }) => ({
+			data,
+			width,
+			height,
+			thumbFile: { ...file, type: mime.lookup(format) || file?.type || '' } as IUpload,
```

Comment:
Idk what thumbFile exactly is, but if it's the "thumbnail" file, it shouldn't fall back to original file's type. That's confusing. While falling back to empty isn't perfect either, but at least we can visually fallback to the original file's mimetype.